### PR TITLE
test(profiles): Automate Profile update UI cases

### PR DESCRIPTION
**Pull request by asimmohx**
## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed. DONE
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency. NO

## What are you changing?
Created new test case for Profile section to create, read, update and delete actions for TLS using real data.

### CODE REVIEW

File: cypress/integration/profile/create-tls.real.ui.spec.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*********************************************************************
+ * Copyright (c) Intel Corporation 2021
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+// Test the create, read, update and delete profile
+
+describe('Create Read Update Delete Profile Test Suite', () => {
```
Comment (rsdmike) [inline cypress/integration/profile/create-tls.real.ui.spec.ts:7]: It looks like we have a create test already as well as a delete test. I would look at those for what to base future tests off of . That said, i think an update test would be useful, but needs to be in its own file and expanded to include api status code checks, mocks, and expectations around all values that change

**Review by rsdmike — CHANGES_REQUESTED** [review]

**Comment by rsdmike** [comment]
Good start, typically check to make sure all the checks are passing before requesting a review in the future. Thanks!

**Comment by asimmohx** [comment]
Sure, Thanks, I will remember that next time

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

File: cypress.json [hunk 2]
```diff
@@ -1,14 +1,15 @@
 {
   "projectId": "x5wpqh",
```
[the remark this answers is not in the document]
Comment (asimmohx) [reply cypress.json:2]: ok

File: src/environments/environment.ts [hunk 3]
```diff
@@ -8,8 +8,8 @@
 
 export const environment = {
```
[the remark this answers is not in the document]
Comment (asimmohx) [reply src/environments/environment.ts:9]: ok

File: cypress/integration/profile/update-tls.spec.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+    cy.myIntercept('GET', 'wirelessconfigs?$count=true', {
+      statusCode: httpCodes.SUCCESS,
+      body: apiResponses.wirelessConfigs.getAll.forProfile.response
+    }).as('get-wirelessConfigs')
+
+    cy.myIntercept('GET', 'profiles?$top=25&$skip=0&$count=true', {
+      statusCode: httpCodes.SUCCESS,
+      body: apiResponses.profiles.getAll.empty.response
+    }).as('get-profiles')
```
Comment (rsdmike) [inline cypress/integration/profile/update-tls.spec.ts:35]: why are we mocking with an empty response? this causes a failure for me.

File: cypress/integration/profile/update-tls.spec.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
+/*********************************************************************
```
Comment (rsdmike) [inline cypress/integration/profile/update-tls.spec.ts:1]: I think lets rename this file to "update.spec.ts" since it tests other updates besides TLS (wireless config, network configuration static) -- its more generic.

**Review by rsdmike — COMMENTED** [review]
These are good enhancements for our UI tests. Thanks! However, they aren't usable for our e2e-test flows.  If it is to be used in our e2e tests, generally, the spec.ts will need to focus on one thing -- updating all the different fields makes it hard to follow which flow we would be testing.

File: cypress/integration/profile/update-tls.spec.ts (hunk 5 above)
Comment (asimmohx) [reply cypress/integration/profile/update-tls.spec.ts:1]: I agree. I thought updating wireless config, network configurations are one-off cases so I included them in the same file. If you think it will be easier to read (maybe extend coverage) in future then I can split into multiple spec file? Or I can just rename the file as you suggested.

File: cypress/integration/profile/update-tls.spec.ts (hunk 4 above)
Comment (asimmohx) [reply cypress/integration/profile/update-tls.spec.ts:35]: Looking, as it might be redundant.

File: cypress/integration/profile/update-tls.spec.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+      statusCode: httpCodes.SUCCESS,
+      body: apiResponses.profiles.getAll.success.response
+    }).as('get-profiles2')
+    cy.myIntercept('GET', 'happyTlspath', {
+      statusCode: httpCodes.SUCCESS,
+      body: apiResponses.profiles.getAll.success.response.data
+    }).as('get-profile')
+  })
+
+  it('Update profile to Admin Control Mode', () => {
```
Comment (rsdmike) [inline cypress/integration/profile/update-tls.spec.ts:51]: Can we combine to a single update test that updates _all_ fields - instead of one field per test?

File: cypress/integration/profile/update-tls.spec.ts [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+      statusCode: httpCodes.SUCCESS,
+      body: profileFixtures.happyPathTls
+    }).as('patch-profile')
+
+    cy.get('mat-row').contains(profileFixtures.happyPathTls.profileName).click()
+    cy.wait('@get-profile')
+    cy.get('mat-select[formcontrolname=activation').click()
+    cy.contains('Admin Control Mode').click({ force: true })
+    cy.setAMTMEBXPasswords(Cypress.env('AMTPASSWORD'), Cypress.env('MEBXPASSWORD'))
+    cy.contains('SAVE').click({ force: true })
```
Comment (rsdmike) [inline cypress/integration/profile/update-tls.spec.ts:62]: why are these  forced?

Comment (asimmohx) [reply cypress/integration/profile/update-tls.spec.ts:62]: I was seeing flaky test runs when I was developing. I can try to remove it now and see if they work.

**Review by matt-primrose — APPROVED** [review]

**Comment by matt-primrose** [comment]
@asimmohx please resolve the conflicts for package.json and package-lock.json.  Should just need to merge main into your fork and update your PR.

**Comment by asimmohx** [comment]
> @asimmohx please resolve the conflicts for package.json and package-lock.json. Should just need to merge main into your branch and update your PR.

Done. Could you please check?

**Comment by asimmohx** [comment]
Accidentally closed the PR so created new PR with the exact same commit,
https://github.com/open-amt-cloud-toolkit/sample-web-ui/pull/632/files#
